### PR TITLE
Add quirks for ts0041, ts0043 and ts0044 (Tuya 1, 3 and 4 button Remotes)

### PR DESCRIPTION
**Pull request by tube0013**
Add Quirk for Tuya 3 button remote
Make TuyaSmartRemote Class a bit more generic

Solves #595

Tested and work with my LoraTap 3 button remote.

### CODE REVIEW

**Comment by dmulcahey** [comment]
Awesome! TY!

File: zhaquirks/tuya/ts0041.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    SHORT_PRESS,
+)
+
+
+class TuyaSmartRemote0041(TuyaSmartRemote):
+    """Tuya 1-button remote device."""
+
+    signature = {
+        # SizePrefixedSimpleDescriptor(endpoint=1, profile=260, device_type=0, device_version=1, input_clusters=[0, 10, 1, 6], output_clusters=[25]))
+        # MODELS_INFO: [("_TZ3400_keyjqthh", "TS0041")],
```
Comment (dmulcahey) [inline zhaquirks/tuya/ts0041.py:27]: Why is this commented out?

File: zhaquirks/tuya/ts0043.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+
+class TuyaSmartRemote0043(TuyaSmartRemote):
+    """Tuya 3-button remote device."""
+
+    signature = {
+        # SizePrefixedSimpleDescriptor(endpoint=1, profile=260, device_type=0, device_version=1, input_clusters=[0, 10, 1, 6], output_clusters=[25]))
+        # SizePrefixedSimpleDescriptor(endpoint=2, profile=260, device_type=0, device_version=1, input_clusters=[1, 6], output_clusters=[])
+        # SizePrefixedSimpleDescriptor(endpoint=3, profile=260, device_type=0, device_version=1, input_clusters=[1, 6], output_clusters=[])
+        # MODELS_INFO: [("_TZ3000_bi6lpsew", "TS0043")],
```
Comment (dmulcahey) [inline zhaquirks/tuya/ts0043.py:31]: Why is this commented out?

File: zhaquirks/tuya/ts0044.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+class TuyaSmartRemote0044(TuyaSmartRemote):
+    """Tuya 4-button remote device."""
+
+    signature = {
+        # SizePrefixedSimpleDescriptor(endpoint=1, profile=260, device_type=0, device_version=1, input_clusters=[0, 10, 1, 6], output_clusters=[25]))
+        # SizePrefixedSimpleDescriptor(endpoint=2, profile=260, device_type=0, device_version=1, input_clusters=[1, 6], output_clusters=[])
+        # SizePrefixedSimpleDescriptor(endpoint=3, profile=260, device_type=0, device_version=1, input_clusters=[1, 6], output_clusters=[])
+        # SizePrefixedSimpleDescriptor(endpoint=4, profile=260, device_type=0, device_version=1, input_clusters=[1, 6], output_clusters=[])
+        # MODELS_INFO: [("", "TS0044")],
```
Comment (dmulcahey) [inline zhaquirks/tuya/ts0044.py:33]: Why is this commented out? and is this how it reports manufacturer?

**Comment by dumpfheimer** [comment]
This is how my device registeres:

  "manufacturer": "_TZ3000_vp6clf9d",
  "model": "TS0044",

This is the device:
https://www.aliexpress.com/item/1005001626323842.html

**Comment by fz1nfx** [comment]
There are also 2 button devices. 

"manufacturer": "_TZ3000_owgcnkrh",
"model": "TS0042",

This device:
[https://www.aliexpress.com/item/4001095272467.html
](url)

**Comment by tube0013** [comment]
> There are also 2 button devices.
> 
> "manufacturer": "_TZ3000_owgcnkrh",
> "model": "TS0042",
> 
> This device:
> [https://www.aliexpress.com/item/4001095272467.html ](url)

Put up a new PR to add this... thanks!

**Comment by oxamafia** [comment]
Hello
Thank you very much for this hard work 
Actually, I have two TS0041 that don’t seem to apply the quirk. I removed them many times, relinked them... it still doesn’t work (lastest HA version)
« Informations sur l’appareil
TS0041
par _TZ3400_keyjqthh
dresden elektronik ConBee II »
Is there a way to force the quirk?

**Comment by loglux** [comment]
https://www.aliexpress.com/item/1005001532044556.html
```
Device info
TS0043
by _TZ3000_bi6lpsew
Zigbee info
IEEE: 84:71:27:ff:fe:13:bc:61
Nwk: 0x3e27
Device Type: EndDevice
LQI: 240
RSSI: -40
Last Seen: 2021-02-12T14:03:22
Power Source: Battery or Unknown
Quirk: zhaquirks.tuya.ts0043.BenexmartRemote0043

```

**Pull request merged**